front_matter/test.ts

```diff
@@ -2,21 +2,50 @@
 
 import { EXTRACT_REGEXP_MAP } from "./_formats.ts";
 
+/** Supported format for front matter. `"unknown"` is used when auto format detection logic fails. */
 export type Format = "yaml" | "toml" | "json" | "unknown";
 
 /**
- * Tests if a string has valid front matter. Supports YAML, TOML and JSON.
+ * Tests if a string has valid front matter.
+ * Supports {@link https://yaml.org | YAML}, {@link https://toml.io | TOML} and
+ * {@link https://www.json.org/ | JSON}.
  *
  * @param str String to test.
  * @param formats A list of formats to test for. Defaults to all supported formats.
+ * @returns `true` if the string has valid front matter, otherwise `false`.
  *
+ * @example
  * ```ts
  * import { test } from "@std/front-matter/test";
+ * import { assert } from "@std/assert/assert";
  *
- * test("---\ntitle: Three dashes marks the spot\n---\n"); // true
- * test("---toml\ntitle = 'Three dashes followed by format marks the spot'\n---\n"); // true
- * test("---json\n{\"title\": \"Three dashes followed by format marks the spot\"}\n---\n"); // true
- * test("---json\n{\"title\": \"Three dashes followed by format marks the spot\"}\n---\n", ["yaml"]); // false
+ * const yamlValid = test(
+ * `---
+ * title: Three dashes marks the spot
+ * ---
+ * `);
+ * assert(yamlValid);
+ *
+ * const tomlValid = test(
+ * `---toml
+ * title = 'Three dashes followed by format marks the spot'
+ * ---
+ * `);
+ * assert(tomlValid);
+ *
+ * const jsonValid = test(
+ * `---json
+ * {"title": "Three dashes followed by format marks the spot"}
+ * ---
+ * `);
+ * assert(jsonValid);
+ *
+ * const parseJsonFrontMatterWithYamlFormat = test(
+ * `---json
+ * {"title": "Three dashes followed by format marks the spot"}
+ * ---
+ * `, ["yaml"]);
+ * assert(!parseJsonFrontMatterWithYamlFormat);
```

Comment:
I think we can separate this into a few small examples. WDYT?

Reply:
Sounds good, it's now done!